### DISCUSSION TITLE
Use “<“ and “>” in an action.conf

### DISCUSSION THREAD

**Discussion by dBsooner** [fail2ban/fail2ban#3582] [Q&A] [answered]
Is it possible to use less than (<) and greater than (>) symbols in an action.conf? I’m trying to setup a Discord notification system posting to webhooks (Discord API). In order for URLs to not show previews on Discord, they have to be wrapped in <>. Which, I assume is strictly for use by fail2ban as variable usage. Is there a way to escape the <>?

**Comment by sebres** [comment, marked as answer]
> Is it possible to use less than (`<`) and greater than (`>`) symbols in an action.conf?

Yes it is... Only available tags `<token>`, where `token` normally contains alphanumeric, underscore and minus characters, will be replaced recursively (so `<primary-<secondary>>` form too. Regex for tokens looks like https://github.com/fail2ban/fail2ban/blob/8eac5f597aaf0c993f4ba006beeeafa28b18bafe/fail2ban/helpers.py#L350
If you use `<` and `>` alone (e. g. separated by spaces from rest of the command, they will be ignored).

> Is there a way to escape the `<``>`?

Yes, one could add something like that in definition or default section:
```ini
[DEFAULT]
lt = <
gt = >
```
And then use `<lt>` or `%(lt)s` for `<` char and `<gt>` or `%(gt)s` for `>` char.
But as already said it is unneeded, if you write it like `cmd 1>some-log 2>&1 <<< "some stdin"` (e. g. with a space delimiter before or after), they will be never considered as part of tags.
The only possible ambiguous way will be something like this - `cmd <file-in>file-out`, because then `file-in` can be considered as a token, if such parameter or variable gets declared in conf-file or its includes. 
However nobody really writes it in this way, rather somehow like this (with a space in between, so fully safe):
 ```diff
- cmd <file-in>file-out
+ cmd <file-in >file-out
```

**Reply by dBsooner** [reply]
Perfect, Thank you!
